examples/js/postprocessing/DepthRangeDofPass.js

```diff
@@ -0,0 +1,147 @@
+  });
+  // this.hBlurMaterial.uniforms.tDiffuse.value = this.normalTarget.texture;
+  this.hBlurMaterial.uniforms.tDiffuse.value = null;
+  this.hBlurMaterial.uniforms.h.value = 1.0 / (resolution.width * blurRatio);
+  //vertical
+  this.vBlurMaterial = new THREE.ShaderMaterial({
+    uniforms: THREE.VerticalBlurShader.uniforms,
+    vertexShader: THREE.VerticalBlurShader.vertexShader,
+    fragmentShader: THREE.VerticalBlurShader.fragmentShader
+  });
+  this.vBlurMaterial.uniforms.tDiffuse.value = this.xblurTarget.texture;
+  this.vBlurMaterial.uniforms.v.value = 1.0 / (resolution.height * blurRatio);
+
+  //depth material
+  this.depthMaterial = new THREE.ShaderMaterial({
+    uniforms: {},
+    vertexShader: THREE.DepthDetectShader.vertexShader,
+    fragmentShader: THREE.DepthDetectShader.fragmentShader
+  });
+
+  //dof materials
+  this.dofMaterial = new THREE.ShaderMaterial({
+    uniforms: {
+      nearZ: { value: null},
+      farZ: { value: null},
+      focusRange: { value: 0 },
+      blurTex: { value: this.yblurTarget.texture },
+      // normalTex: { value: this.normalTarget.texture },
+      normalTex: { value: null },
+      depthTex: { value: this.depthTarget.texture }
+    },
+    vertexShader: THREE.DepthRangeDofShader.vertexShader,
+    fragmentShader: THREE.DepthRangeDofShader.fragmentShader
+  });
+
+  //pps screen quad
+  this.quad2 = new THREE.Mesh(new THREE.PlaneBufferGeometry(2, 2, 64, 64), null);
+  this.quad2.frustumCulled = false;
+  this.scene2.add(this.quad2);
+}
+
+THREE.DepthRangeDofPass.prototype = Object.assign(Object.create(THREE.Pass.prototype), {
+  constructor: THREE.DepthRangeDofPass,
+
+  render: function (renderer, writeBuffer, readBuffer, delta, maskActive) {
+    //render depth target to detect the pixels for blur
+    this.scene.overrideMaterial = this.depthMaterial;
+    renderer.render(this.scene, this.camera, this.depthTarget);
+    this.scene.overrideMaterial = null;
+    // //normal scene, for mix into blur
+    // this.scene.overrideMaterial = null;
+    // renderer.render(this.scene, this.camera, this.normalTarget);
+
+    //out put blur pps screen quad into fbo
+    this.quad2.material = this.hBlurMaterial;
+    renderer.render(this.scene2, this.camera2, this.xblurTarget);
+    //out put blur pps screen quad into fbo
+    this.quad2.material = this.vBlurMaterial;
+    renderer.render(this.scene2, this.camera2, this.yblurTarget);
+
+    // //update focus
+    this.updateFocusZ();
+    //render pps screen
+    this.hBlurMaterial.uniforms.tDiffuse.value = readBuffer.texture;
+    this.dofMaterial.uniforms.normalTex.value = readBuffer.texture;
+
+    this.quad2.material = this.dofMaterial;
+    if (this.renderToScreen) {
+      renderer.render(this.scene2, this.camera2);
+    } else {
+      renderer.render(this.scene2, this.camera2, writeBuffer, this.clear);
+    }
+  }
+});
+
+THREE.DepthRangeDofPass.prototype.updateFocusZ = function() {
+  var camWorldDir = new THREE.Vector3();
```

Comment:
Tip: You are instantiating 5 * 60 = 300 `Vector3` per second by calling `new` and `clone()` in this method. Use a closure to avoid doing that.

Reply:
Hi WestLangley, many thanks for your great advice! Fixed the memory waste in the render loop.
![fix-memory](https://user-images.githubusercontent.com/10095269/40621192-5e615a16-62ce-11e8-8930-d1f9fa65296f.png)